tests/Builder/Analytics/FunctionBuilderTest.php

```diff
@@ -0,0 +1,633 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Solarium\Tests\Builder\Analytics;
+
+use PHPUnit\Framework\TestCase;
+use Solarium\Builder\AbstractExpressionVisitor;
+use Solarium\Builder\Analytics\AnalyticsExpressionVisitor;
+use Solarium\Builder\Analytics\FunctionBuilder;
+use Solarium\Builder\ExpressionInterface;
+use Solarium\Exception\RuntimeException;
+
+/**
+ * FunctionBuilderTest.
+ *
+ * @author wicliff <wicliff.wolda@gmail.com>
+ */
+class FunctionBuilderTest extends TestCase
+{
+    /**
+     * @throws \PHPUnit\Framework\ExpectationFailedException
+     * @throws \Solarium\Exception\RuntimeException
+     *
+     * @see https://lucene.apache.org/solr/guide/analytics.html#example-construction
+     */
+    public function testBuilder(): void
+    {
+        $builder = FunctionBuilder::create()
+            ->where(FunctionBuilder::expr()->div(
```

Comment:
yes, that's another way of doing it and would achieve the same if i'm not mistaken.

the reason for writing it like this in the test and example is purely because i'm used to doing it like that when using doctrine's expression builder (which pattern i used for this and the filter query builder).

just let me know if you want me to outline the both approaches in the documentation part and i'll add another example

Reply:
modified documentation and some of the unit tests to outline te two different usages of expressions